proxy-node-gateway/src/test/java/uk/gov/ida/notification/apprule/base/GatewayAppRuleTestBase.java

```diff
@@ -1,22 +1,103 @@
+import javax.ws.rs.core.NewCookie;
+import javax.ws.rs.core.Response;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.fail;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.TEST_RP_PRIVATE_SIGNING_KEY;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.TEST_RP_PUBLIC_SIGNING_CERT;
 
 public class GatewayAppRuleTestBase {
 
+    private Map<String, NewCookie> cookies;
+
+    protected static final String CONNECTOR_NODE_ENTITY_ID = "http://connector-node:8080/ConnectorResponderMetadata";
+
+    @ClassRule
+    public static final TranslatorClientRule translatorClientRule;
+
+    @ClassRule
+    public static final EidasSamlParserClientRule espClientRule = new EidasSamlParserClientRule();
+
     @ClassRule
-    public static final TranslatorClientRule translatorClientRule = new TranslatorClientRule();
+    public static final VerifyServiceProviderClientRule vspClientRule = new VerifyServiceProviderClientRule();
+
+    static {
+        try {
+            InitializationService.initialize();
+            VerifySamlInitializer.init();
+            translatorClientRule = new TranslatorClientRule();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected final SamlObjectMarshaller marshaller = new SamlObjectMarshaller();
+    protected final Credential countrySigningCredential = new TestCredentialFactory(TEST_RP_PUBLIC_SIGNING_CERT, TEST_RP_PRIVATE_SIGNING_KEY).getSigningCredential();
+    protected final SamlObjectSigner samlObjectSigner = new SamlObjectSigner(
+            countrySigningCredential.getPublicKey(),
+            countrySigningCredential.getPrivateKey(),
+            TEST_RP_PUBLIC_SIGNING_CERT);
 
     @Rule
     public GatewayAppRule proxyNodeAppRule = new GatewayAppRule(
-            ConfigOverride.config("eidasSamlParserService.url", "http://eidas-saml-parser/eidasAuthnRequest"),
-            ConfigOverride.config("verifyServiceProviderService.url", "http://verify-service-provider/"),
+            ConfigOverride.config("eidasSamlParserService.url", espClientRule.baseUri().toString()),
+            ConfigOverride.config("verifyServiceProviderService.url", vspClientRule.baseUri().toString()),
             ConfigOverride.config("translatorService.url", translatorClientRule.baseUri() + "/translator/SAML2/SSO/Response")
     );
+
+    protected Response postEidasAuthnRequest(AuthnRequest eidasAuthnRequest) {
+        System.out.println(marshaller.transformToString(eidasAuthnRequest));
+        String encodedRequest = Base64.encodeAsString(marshaller.transformToString(eidasAuthnRequest));
+        Form postForm = new Form().param(SamlFormMessageType.SAML_REQUEST, encodedRequest);
+
+        Response response = null;
+
+        try {
+            response = proxyNodeAppRule.target("/SAML2/SSO/POST").request().post(Entity.form(postForm));
+        } catch (URISyntaxException e) {
```

Comment:
Why is this exception being specifically caught?

Reply:
Part of a method reinstated; now removed catch block